apps/dev-recruiters/src/pages/dashboard.tsx

```diff
@@ -22,35 +22,32 @@ interface UserData {
   projects?: Project[];
 }
 
-interface Opportunity {
-  readonly id: number;
-  title: string;
-  level: string;
-  roleCategory: string;
-}
-
 const Dashboard: React.FC = () => {
-  const { userData, isAuthenticated } = useUserDataContext();
+  const { userData, isAuthenticated, isLoading } = useUserDataContext();
   const [teamNames, setTeamNames] = useState<string[]>([]);
-  const [activeRoles, setActiveRoles] = useState<Opportunity>([]);
-  const [archivedRoles, setArchivedRoles] = useState<Opportunity>([]);
+  const [activeRoles, setActiveRoles] = useState<Opportunity[]>([]);
+  const [archivedRoles, setArchivedRoles] = useState<Opportunity[]>([]);
   const router = useRouter();
-
-  console.log(userData);
   // roleCategory is the department.
 
   // restrict /dashboard page for project leaders only.
   //some(...).some(...) — checks if any project has a team where the current user (userData.id) is a leader.
 
   useEffect(() => {
+    if (isLoading) {
+      return;
+    }
+
     if (isAuthenticated && userData) {
       const isLeader = userData.projects?.some((project: any) =>
         project.team?.leaders?.some((leader: any) => leader.id === userData.id)
       );
 
-      if (!isLeader) {
-        router.replace('/');
-      }
+      // if (!isLeader) {
```

Comment:
I think we need this to make sure users that are not team leads can access this page.